source/credentials_provider_sts_web_identity.c

```diff
@@ -0,0 +1,1120 @@
+            "(id=%p) STS_WEB_IDENTITY credentials provider query error appending response: %s",
+            (void *)sts_web_identity_user_data->sts_web_identity_provider,
+            aws_error_str(aws_last_error()));
+
+        return AWS_OP_ERR;
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+static int s_sts_web_identity_on_incoming_headers_fn(
+    struct aws_http_stream *stream,
+    enum aws_http_header_block header_block,
+    const struct aws_http_header *header_array,
+    size_t num_headers,
+    void *user_data) {
+
+    (void)header_array;
+    (void)num_headers;
+
+    if (header_block != AWS_HTTP_HEADER_BLOCK_MAIN) {
+        return AWS_OP_SUCCESS;
+    }
+
+    struct aws_credentials_provider_sts_web_identity_user_data *sts_web_identity_user_data = user_data;
+    if (header_block == AWS_HTTP_HEADER_BLOCK_MAIN) {
+        if (sts_web_identity_user_data->status_code == 0) {
+            struct aws_credentials_provider_sts_web_identity_impl *impl =
+                sts_web_identity_user_data->sts_web_identity_provider->impl;
+            if (impl->function_table->aws_http_stream_get_incoming_response_status(
+                    stream, &sts_web_identity_user_data->status_code)) {
+
+                AWS_LOGF_ERROR(
+                    AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+                    "(id=%p) STS_WEB_IDENTITY credentials provider failed to get http status code: %s",
+                    (void *)sts_web_identity_user_data->sts_web_identity_provider,
+                    aws_error_str(aws_last_error()));
+
+                return AWS_OP_ERR;
+            }
+            AWS_LOGF_DEBUG(
+                AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+                "(id=%p) STS_WEB_IDENTITY credentials provider query received http status code %d",
+                (void *)sts_web_identity_user_data->sts_web_identity_provider,
+                sts_web_identity_user_data->status_code);
+        }
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+static void s_sts_web_identity_query_credentials(
+    struct aws_credentials_provider_sts_web_identity_user_data *sts_web_identity_user_data);
+
+static void s_sts_web_identity_on_stream_complete_fn(struct aws_http_stream *stream, int error_code, void *user_data) {
+    struct aws_credentials_provider_sts_web_identity_user_data *sts_web_identity_user_data = user_data;
+
+    aws_http_message_destroy(sts_web_identity_user_data->request);
+    sts_web_identity_user_data->request = NULL;
+
+    struct aws_credentials_provider_sts_web_identity_impl *impl =
+        sts_web_identity_user_data->sts_web_identity_provider->impl;
+    struct aws_http_connection *connection = impl->function_table->aws_http_stream_get_connection(stream);
+    impl->function_table->aws_http_stream_release(stream);
+    impl->function_table->aws_http_connection_manager_release_connection(impl->connection_manager, connection);
+
+    /*
+     * On anything other than a 200, nullify the response and pretend there was
```

Comment:
Are we pretending, or just reporting the error? I ask because if I came in here and found this comment, I might assume we're ignoring/translating something.